src/compiler/checker.ts

```diff
@@ -3187,6 +3187,13 @@ namespace ts {
                 const declaration = symbol.declarations[0];
                 const name = getNameOfDeclaration(declaration);
                 if (name) {
+                    if (name.kind === SyntaxKind.ComputedPropertyName &&
+                        symbol.flags & SymbolFlags.Transient &&
+                        !((symbol as TransientSymbol).checkFlags & CheckFlags.Late) &&
+                        !isWellKnownSymbolSyntactically((name as ComputedPropertyName).expression) &&
+                        symbol.escapedName) {
+                        return '["' + unescapeLeadingUnderscores(symbol.escapedName) + '"]';

```

Comment:
Previously this would print `[0]` if the computed property originally was a numeric literal. I know this is more correct with how we view the type (since keys are always strings), but it does feel unfortunate to be always quoting everything in the result. Maybe not quoting numeric names here would make me feel better? It would probably also reduce baseline noise.

Reply:
True on all points. It's easy enough to change, so I did. But as you noticed below, we aren't *really* trying to preserve what the user typed with this change.